tests/integ/test_udtf.py

```diff
@@ -687,6 +687,240 @@ def end_partition(self, df: pd.DataFrame) -> pd.DataFrame:
     assert_vectorized_udtf_result(session.table(vectorized_udtf_test_table), my_udtf)
 
 
+@pytest.mark.parametrize("from_file", [True, False])
+def test_register_vectorized_udtf_process(session, from_file, resources_path):
+    data = [
+        Row("x", 3, 35.9),
+        Row("x", 9, 20.5),
+        Row("x", 12, 93.8),
+        Row("y", 5, 69.2),
+        Row("y", 10, 94.3),
+        Row("y", 15, 36.9),
+        Row("y", 20, 85.4),
+        Row("z", 10, 30.4),
+        Row("z", 20, 85.9),
+        Row("z", 30, 63.4),
+        Row("z", 40, 35.8),
+        Row("z", 50, 95.4),
+    ]
+    df = session.create_dataframe(data, schema=["id", "col1", "col2"])
+
+    class BasicProcess:
+        def process(self, df):
+            return df
+
+    output_schema = PandasDataFrameType(
+        [StringType(), IntegerType(), FloatType()], ["_id", "_col1", "_col2"]
+    )
+    input_types = [PandasDataFrameType([StringType(), IntegerType(), FloatType()])]
+    if from_file:
+        process_udtf = session.udtf.register_from_file(
+            TestFiles(resources_path).test_vectorized_udtf_py_file,
+            "BasicProcess",
+            output_schema=output_schema,
+            input_types=input_types,
+        )
+    else:
+        process_udtf = session.udtf.register(
+            BasicProcess, output_schema=output_schema, input_types=input_types
+        )
+    Utils.check_answer(df.select(process_udtf("id", "col1", "col2")), data)
```

Comment:
IMO it would be better style to break this test up into several after each of these check_answers. That way it is more clear which cases are actually breaking when the test fails.